src/opustrainer/modifiers/typos.py

```diff
@@ -0,0 +1,152 @@
+import random
+import re
+from typing import Dict, Literal, Tuple
+
+import typo
+
+from opustrainer.modifiers import Modifier
+
+
+def random_space_with_alignment(action:Literal['random_space', 'skipped_space'], strval:str, alignments:str) -> Tuple[str, str]:
+    """Special version of typo's random_space and skipped_space that also
+    updates alignment info.
+    
+    action: add | remove
+      whether to add or remove a random space
+
+    strval: str
+      input text
+
+    alignments: str
+      string of space split m-n pairs
+
+    """
+    # all the locations where there are non-space characters.
+    locations = [m.start() for m in re.finditer(r'\S', strval)]
+    
+    if len(locations) == 0:
+        return strval, alignments
+
+    # Select character after which to add a space
+    char_index = locations[random.randint(0, len(locations) - 1)]
+
+    # Figure out which word that character falls in
+    word_index = sum(1 for _ in re.finditer(r'\s+', strval[:char_index]))
+
+    # Insert space
+    if action == 'random_space':
+        strval = strval[:char_index] + ' ' + strval[char_index:]
+    else:
+        strval = strval[:char_index-1] + strval[char_index:]
+
+    # Fix up alignments
+    fixed_alignments = []
+    for alignment in alignments.split(' '):
+        # Splits the a-b pairs into tuples
+        src, trg = [int(index) for index in alignment.split('-', maxsplit=1)]
+
+        # Alignments before the introduced space stay as-is. Intentionally, if
+        # the mapping is about word_index itself, we apply both to duplicate
+        # the mapping.
+        if src <= word_index:
+            fixed_alignments.append(f'{src}-{trg}')
+        
+        # Alignments after the space are shifted by 1
+        if action == 'random_space' and src >= word_index \
+           or action == 'skipped_space' and src > word_index:
+            src += 1 if action == 'random_space' else -1
+            fixed_alignments.append(f'{src}-{trg}')
+
+    return strval, ' '.join(fixed_alignments)
+
+
+class TypoModifier(Modifier):
+    modifiers = {
```

Comment:
There's 9 modifiers here,so the overall probability mass is 90% when using default settings. Where does the other 10% go?